### PR TITLE
Fixes #24270: Handle server errors

### PR DESCRIPTION
**Pull request by akofink**
Don't update the cache if the server fails to update

### CODE REVIEW

File: src/katello/repos.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def report_enabled_repos(consumer_id, report):
     uep = get_uep()
     method = '/systems/%s/enabled_repos' % uep.sanitize(consumer_id)
     try:
-        uep.conn.request_put(method, report)
+        return uep.conn.request_put(method, report)
```
[the remark this answers is not in the document]
Comment (akofink) [reply src/katello/repos.py]: I guess, my one last effort - would you be making this argument in Ruby? ;) If you don't return anything, just like here, it's `nil`. I think it's a useful language feature that we don't always need boolean values. The return, by the way, is the put response from [`urllib.request`](https://docs.python.org/3/library/urllib.request.html), which could be useful in the future.

[the remark this answers is not in the document]
Comment (akofink) [reply src/katello/repos.py]: Also, Ruby makes this easier to look at, perhaps:

```rb
def foo
  return :yay
rescue Exception
  handle_exception
end
```

**Comment by akofink** [comment]
Just noticed, you asked for explicit False from the exception. That's not what I did, are you ok with the explicit True, or do you want two return statements?

**Comment by jlsherrill** [comment]
@akofink i think i'm okay with either, as each makes it more obvious to me whats going on

**Pull request merged**